robottelo/cli/factory.py

```diff
@@ -2531,3 +2536,215 @@ def setup_org_for_a_rh_repo(options=None):
+            'dns-id': puppet_proxy['id'],
+            'tftp-id': puppet_proxy['id'],
+        })
+    else:
+        # Create new subnet
+        subnet = make_subnet({
+            'name': gen_string('alpha'),
+            'network': network,
+            'mask': settings.vlan_networking.netmask,
+            'domain-ids': domain['id'],
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dhcp-id': puppet_proxy['id'],
+            'dns-id': puppet_proxy['id'],
+            'tftp-id': puppet_proxy['id'],
+        })
+
+    # Search if Libvirt compute-resource already exists. If so, just update its
+    # relevant fields otherwise, create new compute-resource with 'libvirt'
+    # provider.
+    current_libvirt_url = (
+        LIBVIRT_RESOURCE_URL % settings.compute_resources.libvirt_hostname
+    )
+
+    libvirt_resources = [
+        ComputeResource.info({'id': comp_res['id']}) for comp_res
+        in ComputeResource.list()
+    ]
+    comp_resources = [
+        comp_res for comp_res in libvirt_resources
+        if comp_res['url'] == 'url={0}'.format(current_libvirt_url) and
+        comp_res['provider'] == 'Libvirt'
+    ]
+    if len(comp_resources) >= 1:
+        ComputeResource.update({
+            u'id': comp_resources[0]['id'],
+            u'location-ids': loc['id'],
+            u'organization-ids': org['id'],
+        })
+    else:
+        # Create Libvirt compute-resource
+        make_compute_resource({
+            'name': gen_string('alpha'),
+            'provider': 'libvirt',
+            'url': current_libvirt_url,
+            'set-console-password': False,
+            'display-type': 'VNC',
+            u'location-ids': loc['id'],
+            u'organization-ids': org['id'],
+        })
+
+    # Get the Partition table entity
+    ptable = PartitionTable.info({'name': DEFAULT_PTABLE})
+
+    # Get the OS entity
+    os = OperatingSys.list({
+        'search': 'name="RedHat" AND major="{0}" OR major="{1}"'.format(
+            RHEL_6_MAJOR_VERSION, RHEL_7_MAJOR_VERSION)
+    })[0]
+
+    # Get proper Provisioning templates and update with OS, Org, Location
+    provisioning_template = Template.info({
+        'name': 'Satellite Kickstart Default'})
+    Template.update({
+        u'id': provisioning_template['id'],
+        u'operatingsystem-ids': os['id'],
+        u'location-ids': loc['id'],
+        u'organization-ids': org['id'],
+    })
+    pxe_template = Template.info({
+        'name': 'Kickstart default PXELinux'})
+    Template.update({
+        u'id': pxe_template['id'],
+        u'operatingsystem-ids': os['id'],
+        u'location-ids': loc['id'],
+        u'organization-ids': org['id'],
+    })
+
+    # Get the architecture entity
+    arch = Architecture.list(
+        {'search': 'name={0}'.format(DEFAULT_ARCHITECTURE)})[0]
+
+    # Get the media and update its location
+    media = Medium.list({u'organization-id': org['id']})[0]
```

Comment:
Another vulnerable place - what if organization doesn't contain any medium? I've run one test locally and it failed at this line as no medium were found.